raft/storage.go

```diff
@@ -41,7 +41,9 @@ type Storage interface {
 	// InitialState returns the saved HardState and ConfState information.
 	InitialState() (pb.HardState, pb.ConfState, error)
 	// Entries returns a slice of log entries in the range [lo,hi).
-	Entries(lo, hi uint64) ([]pb.Entry, error)
+	// MaxSize limits the total size of the log entries returned, but
+	// Entries returns at least one entry if any.
+	Entries(lo, hi, maxSize uint64) ([]pb.Entry, error)
```

Comment:
We can do length too if you like. etcd does not care about this too much, since most of the entries are very small in size and we have a very good idea on the avg size.